cylc/uiserver/websockets/tornado.py

```diff
@@ -58,19 +60,25 @@ async def _handle(self, ws, request_context):
         await self.on_open(connection_context)
         pending = set()
         while True:
+            message = None
             try:
                 if connection_context.closed:
                     raise ConnectionClosedException()
-                message = await connection_context.receive()
+                message = connection_context.receive()
             except ConnectionClosedException:
                 break
+            except QueueEmpty:
+                pass
             finally:
                 if pending:
                     (_, pending) = await wait(pending, timeout=0, loop=self.loop)
 
-            task = ensure_future(
-                self.on_message(connection_context, message), loop=self.loop)
-            pending.add(task)
+            if message:
+                task = ensure_future(
+                    self.on_message(connection_context, message), loop=self.loop)
```

Comment:
Should this be changed to use `create_task`? (see Python docs: https://docs.python.org/3/library/asyncio-task.html#creating-tasks)

Reply:
(kind of off-topic, sorry, but I was trying to understand what this does :grin: )

Reply:
>Important
>
>See also the create_task() function which is the preferred way for creating new Tasks. 

Doesn't hurt to change to the latest recommended way. Let me change it and quickly test it. :+1:

Reply:
Done!